### ISSUE TITLE
Links are invisible on light colour schemes

### ISSUE THREAD

**Issue by sixhobbits** [squidfunk/mkdocs-material#3958]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

I have tried to check the above four as much as possible, but I have not managed to fully reproduce a minimal example of this problem. We have seen it across a few sites since we upgraded the mkdocs-material version (without making other changes to the CSS etc, so I am fairly confident it is something that changed in MkDocs Material itself).

Links cannot be seen on light themes / when the background is white. Links show up when

* A user mouses over them 
* When a dark theme is enabled

<img width="744" alt="image" src="https://user-images.githubusercontent.com/2641205/170717228-ae0e4bc9-1ab0-43fe-b3da-94b366b9d0b2.png">
<img width="648" alt="image" src="https://user-images.githubusercontent.com/2641205/170717237-6f687721-983c-4029-a5d5-d67f7d5ccb71.png">
<img width="785" alt="image" src="https://user-images.githubusercontent.com/2641205/170717285-8b228216-99ac-47bb-a6de-8937d7e08192.png">

### Expected behaviour

Links can be seen by all users, regardless of theme.

### Actual behaviour

Links cannot be seen, unless a dark theme is enabled.

### Steps to reproduce

You can see this in action currently at https://codecapsules.io/docs/comparisons/ and all other links across the site.

### Package versions

- Python: `python --version`: Python 3.9.12
- MkDocs: `mkdocs --version`: mkdocs, version 1.3.0
- Material: `pip show mkdocs-material | grep -E ^Version`  8.2.15+insiders.4.16.0

### Configuration

```yaml
I have tried and can't do this right now. I will keep working on it, but maybe you already know the answer :)
```

### System information

MacOS

FireFox and Safari

**Comment by facelessuser** [comment]
Configuration is very important. Are you using custom CSS? If so, that is the first thing I'd check.

Considering that the official MkDocs Material documentation is generated with the project itself and does not suffer from invisible links on light color schemes, it seems to be most likely a local issue and not an issue with the project.

**Comment by squidfunk** [comment]
Thanks for reporting. This can be reproduced by going to the [primary color](https://squidfunk.github.io/mkdocs-material/setup/changing-the-colors/#primary-color) guide an clicking on `white`. It's most likely a regression that should be straight-forward to fix, possibly due to specificity issues.

**Comment by squidfunk** [comment]
Fixed in [`221a869`](https://github.com/squidfunk/mkdocs-material-insiders/commit/221a8693ad096d08f2a8cf9dce0e063b030b707a). I've taken the time and refactored the tints of links for each scheme, `default` and `slate`.

- For the `default` theme, the lighter colors like `yellow` or `amber` now define slightly darker link colors
- For the `slate` theme, the darker colors like `indigo` and `teal` now define slightly lighter link colors
- For both themes, neutral colors like `grey` or `white` will always define appropriate tints from the `indigo` palette

This should reduce problems with combining different primary colors with light and dark color schemes. Note that there might still be cases where colors don't work well, but it's just too many combinations to optimize. I think we now have reasonable defaults in-place. Furthermore, I tried to flatten the selectors, which should lead to fewer specificity issues, especially when customizing.

This change is currently only in Insiders but will be merged into the community edition in order to minimize friction. If all is fine with this change, I'll merge it when the next funding goal is reached (maybe mid-June).
